rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -244,35 +256,45 @@ def decode_by_address_rules(
         """
         mapping_result = self.rules.address_mappings.get(tx_log.address)
         if mapping_result is None:
-            return None, []
-        method = mapping_result[0]
+            return None, [], None
+
+        decoder_method, counterparty = mapping_result
+        method = decoder_method[0]
 
         try:
-            if len(mapping_result) == 1:
+            if len(decoder_method) == 1:
                 result = method(tx_log, transaction, decoded_events, all_logs, action_items)
             else:
-                result = method(tx_log, transaction, decoded_events, all_logs, action_items, *mapping_result[1:])  # noqa: E501
+                result = method(tx_log, transaction, decoded_events, all_logs, action_items, *decoder_method[1:])  # noqa: E501
         except (DeserializationError, ConversionError, UnknownAsset) as e:
             log.debug(
                 f'Decoding tx log with index {tx_log.log_index} of transaction '
                 f'{transaction.tx_hash.hex()} through {method.__name__} failed due to {str(e)}')
-            return None, []
+            return None, [], None
 
-        return result
+        return result + (counterparty,)
 
     def run_all_post_decoding_rules(
             self,
             transaction: EvmTransaction,
             decoded_events: list[HistoryBaseEntry],
             all_logs: list[EvmTxReceiptLog],
+            counterparties: set[str],
     ) -> list[HistoryBaseEntry]:
         """
-        Runs all post-decoding rules from self.rules.post_decoding_rules.
+        Runs only the required post decoding rules for this transaction
         The post-decoding rules list consists of tuples (priority, rule) and must be sorted by
         priority in ascending order. The higher the priority number the later the rule is run.
         """
-        for (_, rule) in self.rules.post_decoding_rules:
-            decoded_events = rule(transaction=transaction, decoded_events=decoded_events, all_logs=all_logs)  # noqa: E501
+        # first include the rules that need to be executed always
+        rules = self.rules.post_decoding_rules.common
```

Comment:
there should never be such rules. The entire point of this change is to avoid their existence.